File: server/src/main/java/com/fluffy/oauth2/infra/google/dto/GoogleUserInfoResponse.java
package com.fluffy.oauth2.infra.google.dto;

import com.fasterxml.jackson.databind.PropertyNamingStrategies.SnakeCaseStrategy;
import com.fasterxml.jackson.databind.annotation.JsonNaming;
import com.fluffy.oauth2.domain.OAuth2UserInfo;

@JsonNaming(SnakeCaseStrategy.class)
public record GoogleUserInfoResponse(
        String id,
        String email,
        String name,
        String picture
) {

    public OAuth2UserInfo toOAuth2UserInfo() {
        return new OAuth2UserInfo(
                id,
                name,
                email,
                picture
        );
    }
}

File: server/src/main/kotlin/com/fluffy/oauth2/infra/google/response/GoogleUserInfoResponse.kt
package com.fluffy.oauth2.infra.google.response

import com.fasterxml.jackson.databind.PropertyNamingStrategies.SnakeCaseStrategy
import com.fasterxml.jackson.databind.annotation.JsonNaming
import com.fluffy.oauth2.domain.OAuth2UserInfo

@JsonNaming(SnakeCaseStrategy::class)
data class GoogleUserInfoResponse(
    val id: String,
    val email: String,
    val name: String,
    val picture: String
) {

    fun toOAuth2UserInfo(): OAuth2UserInfo {
        return OAuth2UserInfo(
            id,
            name,
            email,
            picture
        )
    }
}